### ISSUE TITLE
Modifying headers in `buildRequest` causes fetch on every page resize

### ISSUE THREAD

**Issue by aaronschwartz** [heroku/react-refetch#171]
I have a situation where I want to send a token back in the headers with every request. That token is subject to change throughout the life of the app. I want to put it in my helper for setting defaults so that I don't have to worry about it throughout my code.

If I do:
```
connect.defaults({
  headers: {
    "token": localStorage.getItem("token")
  }
})
```
then the token is not updated if it changes because it is only calculated once.

If I add it in `buildRequest`:
```
connect.defaults({
  buildRequest: (mapping) => {
    const options = {
      ...
      headers: mapping.headers
    }

    options.headers["token"] = localStorage.getItem("token");

    return new Request(mapping.url, options);
  }
})
```
then my requests are correct and the token is updated on every call, but my issue is that every time I resize my page it fires a component update, and when it does a header comparison it decides that it has changed (even if the token has not changed) and causes many fetches while resizing.

It looks like it is comparing the header before the `buildRequest` without the token and the header after the `buildRequest` with the token during it's equality check.

It feels like it should save the `[ 'value', 'url', 'method', 'headers', 'body' ]` after the buildRequest to use for comparison purposes to see if the request has changed.

Is there a better way to do what I'm looking for?

**Comment by ryanbrainard** [comment]
Is this fixed by #172?

**Comment by aaronschwartz** [comment]
Yep it should be. Thanks.
